packages/playwright/src/transform/esmLoader.ts

```diff
@@ -72,8 +72,14 @@ async function load(moduleUrl: string, context: { format?: string }, defaultLoad
   const transformed = transformHook(code, filename, moduleUrl);
 
   // Flush the source maps to the main thread, so that errors during import() are source-mapped.
-  if (transformed.serializedCache)
-    await transport?.send('pushToCompilationCache', { cache: transformed.serializedCache });
+  // Under certain conditions with ESM -> CJS -> CJS imports, we can enter deadlock awaiting the
+  // MessagePort transfer simultaneously with the Node.js worker thread that is performing the load().
+  // Purposefully do not await
+  if (transformed.serializedCache) {
+    transport?.send('pushToCompilationCache', { cache: transformed.serializedCache })
```

Comment:
Perhaps we can not await only when `format` is `commonjs` or something like that?